pmd-core/src/main/java/net/sourceforge/pmd/cli/PmdParametersParseResult.java

```diff
@@ -44,6 +44,14 @@ public boolean isHelp() {
         return !isError() && result.isHelp();
     }
 
+    /**
+     * Returns whether parsing just requested the {@code --version} text.
+     * In this case no configuration is produced.
```

Comment:
This is not actually the case, you need to modify `toConfiguration` below to prevent a configuration from being produced